### ISSUE TITLE
feature-request: possibility to change the size of description popup

### ISSUE THREAD

**Issue by steff123** [microsoft/vscode#29126]
Hello, I am creating a snippet for VS Code with long descriptions. I have noticed that the description popup window in VS Code has a scrollbar, but it would be greater if I can enlarge the window. This seems not to be possible in the current version (1.13.1). It would be better if one can either set the height and width in the preferences or change the size of the popup with mouse dynamically.

Regards steff

**Comment by mike-lischke** [comment]
I'd like to add that wrapping in the popup is also a problem and should be avoided (overflow should be horizontally scrollable).

**Comment by ramya-rao-a** [comment]
cc @Microsoft/vscode for thoughts

**Comment by stevencl** [comment]
It seems like the problem is that to read long descriptions, you need to scroll. One way to solve that problem without requiring intervention from the user would be to resize the popup dynamically based on the content it needs to display. If we were able to do this, we could make long descriptions more readable without requiring the user to either scroll through the description or to drag out the size of the window to make the content more readable. We currently set the height dynamically, perhaps we could also look at dynamically altering the width to accommodate larger content?

While there would probably still be cases where there simply isn't enough room to show all of the content this would be the preferable solution for me since it allows most users to keep their hand on the keyboard and means that we don't need to worry about what to do with the popup after the user has resized it, then dismissed it (do we keep it at that size, even if it is way too big for the next piece of content, what happens then when the user returns to the same content that caused them to resize etc etc).

I also agree that wrapping inside the popup doesn't seem like the right thing to do since it alters the readability of the content. So if resizing the width of the popup still means content is outside of the viewport, we should add a scroll bar as @mike-lischke suggests.

**Comment by ramya-rao-a** [comment]
The scrollbar solution is worth pursuing

**Comment by nsreenath** [comment]
Large descriptions that usually come as part of a well documented sample code will be easy to read if we put a button to open the entire description in new tab.

**Comment by Raydir** [comment]
I'd love to get a possibility to "resize" this window. Prefered solution would be a setting like "editor.SuggetionsHeigth" and "editor.SuggetionsWidth" which could be set in pixels.

**Comment by lonix1** [comment]
Like I said in my other issue:

Please add one or more of the following:
- Add a setting for "wider IntelliSense popups"
- Don't wrap
- Or the best option: allow me to resize the popup horizontally, and automatically remember the width

This should be for autocompletion, intellisense and debugging popups (I'm unsure if those are the correct names for vscode). They are all too narrow, and they all wrap. It should be like in regular Visual Studio - we shouldn't have to debate it, just copy what already works.

**Comment by lin1000** [comment]
This is really important, otherwise I always want to "leave" vscode

**Comment by finalcreator** [comment]
when the import package path is very long , the suggestion window width not fit the long length, characters are truncated!!!

**Comment by garfunkel** [comment]
I'd like this feature too - I have to use large font size in my editor, but it makes the popup window dramatically too small in relation to the text size, rendering it ugly/useless.

**Comment by lonix1** [comment]
Interesting point: one of the most compelling reasons IDEs became popular was that they are aware of what you're doing - i.e. intellisense. And *vscode ALREADY HAS this available* (and it's very good!), it's just unusable.

So right now, it's hard to say that using vscode is better than using notepad.

Since intellisense is already implemented and working, I don't understand why it isn't made pretty so it becomes usable. This is a very old issue, and affects everyone.

**Comment by RuiffCardoso** [comment]
After using VSCode for Java for a few months, one of the things I find missing is a better parameter info. When there is a rather large function (long function name or parameters), it gets hard to navigate through the window because they ending part get cut, as seen on the attached picture:

![image](https://user-images.githubusercontent.com/39524345/47366138-eea4f900-d6d4-11e8-9c5f-60a9224ef9b3.png)

![image](https://user-images.githubusercontent.com/39524345/47366318-5c512500-d6d5-11e8-9d78-041e44e07f4c.png)

Having an adjustable window, or even a configured fixed size width, would improve code navigation alot.

**Comment by dariomel** [comment]
In addition to the improvements suggested by lonix1 on Jul 19, 2018, I suggest also adding a horizontal scroll bar that automatically shows up when any items in the list overflow the width of the intellisense window, because even if the window could be resized, there might still be some function signatures that do not fit in the window.

**Comment by mike-lischke** [comment]
@dariomel This is what I suggested already in my first comment (directly after the issue description above).

**Comment by dariomel** [comment]
sorry, I missed that.
I edited my comment accordingly.

Dario


________________________________

**Comment by lonix1** [comment]
Related:

When a function overloads popup shows "1/4" and I click the up arrow then wrap around to "4/4", and if it shows "4/4" and I click the down arrow then show "1/4" - don't just close the popup

**Comment by tracksaw** [comment]
The display length (i.e., the number of items displayed in the list) of the suggestions list is hard-coded to 12 in the maxSuggestionsToShow variable in the src/vs/editor/contrib/suggest/suggestWidget.ts file.
We generally want a longer visible list than this. 
And, yes, it would be good if the window were resizable.

Feature Requested:
Make the display length (maxSuggestionsToShow) of the suggestions list dynamically settable in configuration.

**Comment by NTaylorMullen** [comment]
This would definitely be nice to have!

**Comment by MrMovl** [comment]
I am using vs code for my elm projects and the small suggestion popup makes the language support practically useless. And that is sad, because the elm compiler is so polite and really wants to help me, but it needs more words then vs code allows.

This is what I can see from the editor:
![vs code suggestion box](https://user-images.githubusercontent.com/3818901/53567533-c6844580-3b5f-11e9-9e63-11bbe8e926ba.png)

And this is the full message:
![elm compiler message full](https://user-images.githubusercontent.com/3818901/53567546-d0a64400-3b5f-11e9-8098-df62364868d6.png)


So please, let us modify the suggestion modal size. :pray:

**Comment by kamil-tekiela** [comment]
https://github.com/bmewburn/vscode-intelephense/issues/291

**Comment by tamuratak** [comment]
@MrMovl, in [69388](https://github.com/Microsoft/vscode/issues/69388#issuecomment-467614946), hovers for problem become to have larger width. You can check that in VSCode Insiders now.

**Comment by ReasonableDeveloper** [comment]
Auto resize or the ability to configure the popup would be great. I'm trying to read the intellisense of vscode `keybindings.json` but all the options are cut off.

Upon hovering over the name, I also don't see the full property name, nor a description. The hovering is probably another issue altogether but it does make it harder to deal with this one.

![image](https://user-images.githubusercontent.com/28842311/61573029-ff3da080-aaa7-11e9-9c7f-c4c5dc4fe7a2.png)

**Comment by stmax82** [comment]
I just got referred here from the vscode-python repo:
https://github.com/microsoft/vscode-python/issues/8036

This discussion about making windows resizable seems to be already going on for... 2 and a half years? A quick fix letting users manually resize these windows would be completely fine for now.

**Comment by octref** [comment]
From @DanTup in https://github.com/microsoft/vscode/issues/76855

---

> Raised at https://github.com/Dart-Code/Dart-Code/issues/1837, if the completion items are really long, VS Code truncates them at the end:

> <img width="648" alt="Screenshot 2019-07-07 at 20 31 34" src="https://user-images.githubusercontent.com/3372058/60771916-29ca4980-a0f7-11e9-8240-21c34c790229.png">

> I'll see if we can change this completion so it doesn't include the full path, but it seems like VS Code could maybe be smart here - the visible characters of all the completions here are the same thing as is visible in the document (which is being used to filter) so it would be much better to truncate from the start than the end (for ex., if the label starts with the text being used to filter, then truncate from the start).

**Comment by wompeter** [comment]
There's still no way to do this?  Debugging any reasonably sized object is a really bad experience because of the tiny popups...

**Comment by breathe** [comment]
Want to pile in on this -- my apollo auto-generated typescript types are useless because can't disambiguate the suggestions.

**Comment by awulkan** [comment]
What's the status on this. The popup window is useless right now. Everything gets split into multiple lines randomly, even in the middle of words! It's very hard to read anything inside of the popup.

![https://i.imgur.com/5dVnovM.png](https://i.imgur.com/5dVnovM.png)

**Comment by Srekel** [comment]
I'm also wondering about this - it's a pain writing in C because the function names don't fit. Can we please get setting for making it wider? Or just increasing the default width? What's the point of it not being wide enough to show what's needed?

![image](https://user-images.githubusercontent.com/1578924/79472513-3a9fd800-8004-11ea-8c0a-8ee7098b8722.png)

**Comment by tamuratak** [comment]
My PR #69363 is closed by @jrieken. I am not sure what the exact reason is. Someone who wants to fix this issue can make a new PR based on my PR, communicating with him.

**Comment by brokenthorn** [comment]
My best guess on how vscode should handle this is just do it like other widely known IDEs do it: make the pop-up resize-able from the lower-right corner and remember the size as an editor preference.

**Comment by nickcorona** [comment]
Lol, let's wait another 2 years.

**Comment by lygstate** [comment]
anyidea to resolve this issue?

**Comment by garfunkel** [comment]
I've concluded at this point that this is impossible. :(

**Comment by awulkan** [comment]
It's definitely possible, it's just that the maintainers of this repository apparently don't care. Solutions have been submitted and then shut down.

**Comment by lygstate** [comment]
Why not impossible? any idea about implement this?

**Comment by awulkan** [comment]
@lygstate Well there was this PR: https://github.com/microsoft/vscode/pull/69363
But it got closed and then they removed the ability to discuss it.

**Comment by garfunkel** [comment]
I detect a conspiracy. This is an un-feature.

**Comment by lygstate** [comment]
So, MSft dit that intensionly?

**Comment by povilasbaranovas** [comment]
**This is not a proper solution, but a hack that might break after any program update!**

As VsCode is styled mainly using CSS to change the size you have to simply override default values.
This can be accomplished by using "Custom CSS and JS loader" extension and injecting your own CSS. 

_Width value is static 430px:_ 
`.monaco-editor .suggest-widget {
  width: 430px; 
} `

 _Height value is set inline on element by editor depending on how many suggestions there are so !important is necessary to owerride it:_ 
`.monaco-editor .suggest-widget>.tree {
  height: 300px !important;
}  `

**Comment by DavidGoldman** [comment]
Any update here? This is a big pain, especially with Swift/SourceKit-LSP.

**Comment by jerclarke** [comment]
What's even going on here? Seems like a clear problem was identified and a lot of solutions were proposed. 

I seriously have trouble imagining why anyone would prefer the current behavior. If someone actually uses this box, they probably want it to be bigger and to have control over the size of text. 

I didn't read every single comment, so I'll just add this: The `"terminal.integrated.fontSize"` setting allows us to control the size of text in the left-part of the suggestion box, so why not have one for the right-part?

Please VS Code, fix this! I'm new here and this issue is bordering on dealbreaker, it makes me want to go back to Netbeans *Shudder*

**Comment by jerclarke** [comment]
Alright everyone, I got really excited about @povilasbaranovas proposed solution using the [Custom CSS and JS Loader](https://marketplace.visualstudio.com/items?itemName=be5invis.vscode-custom-css) extension, but when I tried the code, it didn't do what I was hoping. 

I spent a few hours on it and figured out a set of CSS that works pretty well! Things are still a bit jumpy sometimes, but I have my editor showing my full PHPDoc for functions now which is what I wanted. 

[Get the configuratble CSS and other details on this StackOverflow thread I created about this subject](https://stackoverflow.com/a/62963539/209859)

The Details box expands to fit as much as it can, both horizontally and vertically, and the CSS uses variables so you can easily configure the widths and heights to your taste (or depending on your code). 

Here's a demo of how my custom CSS looks out of the box:

<img width="1355" alt="Screen Shot 2020-07-18 at 7 53 20 PM" src="https://user-images.githubusercontent.com/139995/87864703-410a5c80-c931-11ea-8906-718b9783a47d.png">

And here's how that same code looks by default:

<img width="788" alt="Screen Shot 2020-07-18 at 8 01 07 PM" src="https://user-images.githubusercontent.com/139995/87864727-6e570a80-c931-11ea-9548-828b0ebcb369.png">

If you want to use it or improve it please do. If this bug ever gets fixed then someone please update [my SO post](https://stackoverflow.com/a/62963539/209859) too ;)

**VS CODE TEAM**: Please make this obsolete!!!

**Comment by hacantorcooke** [comment]
Just to pile on here. Was thinking of switching my whole team to VS code due to many of the awesome abilities. Will be a no-go until this is sorted. Our code uses long, descriptive names in C which are just cut off by the pop-up, making it effectively useless. Kinda crazy that something so simple has been open for so long; there could be a lot of converts waiting on this.

Related: If you want to keep the box small, something more clever would be to display what makes the options different. I.E. today, an enum list (C) like
APP_HELLOWORLD_TYPE_A
APP_HELLOWORLD_TYPE_B

Will show 
APP_HELLOWORLD_TYPE....
APP_HELLOWORLD_TYPE....

With a little magic you could have it show
A
B
Ideally with them being under a dropdown of sorts labelled APP_HELLOWORLD_TYPE

**Comment by clankill3r** [comment]
I really don't get how this is still an open issue from 2017. This is such a pain in the ***, and affects my daily workflow.
Specially with an editor build around css this should not be an issue that takes more then 3 years to fix.

**Comment by garfunkel** [comment]
OMG it looks like this is getting some attention. Hell yes hell yes hell yes!

**Comment by jrieken** [comment]
Ok, so this is the plan 

* make the completions-list resizable
* make the details-pane resizable
* don't shrink completions-list when details-pane shows

**Comment by jerclarke** [comment]
What happens if it's too big for the window? Details fall down below completions?

**Comment by jrieken** [comment]
FYI - I have merged the first cut into master (https://github.com/microsoft/vscode/pull/109094). This means next insiders (tomorrow) will have this. There might still be rough edges and bugs but I want to get feedback rather sooner than later.  

👉  try this in https://code.visualstudio.com/insiders/ 👈

**Comment by misolori** [comment]
@jrieken these updates are 🔥 . Should there be a min-height on the description that caps where the text ends? This feels a bit odd:

![Kapture 2020-10-22 at 09 13 20](https://user-images.githubusercontent.com/35271042/96900473-ff64e280-1446-11eb-81d4-ebba0526aa36.gif)

Also, it appears that the description height doesn't persist even though all others do?

**Comment by jerclarke** [comment]
It's possible for the text to be extremely tall, so a min-height couldn't be based on that. 

Any chance this update could also give us an option to control the size of the text in these boxes separate from everything else?

**Comment by MartinGC94** [comment]
Maybe it's an extension issue, but I don't see linebreaks for the different method overloads in Powershell. Here's how it looks in stable VS code:
![image](https://user-images.githubusercontent.com/42123497/96933507-dda84180-14c0-11eb-86ad-db3357a7850d.png)

And here's how it looks in the latest insider version:
![image](https://user-images.githubusercontent.com/42123497/96933736-4b546d80-14c1-11eb-954d-00f9109cd3d8.png)

**Comment by jrieken** [comment]
> Any chance this update could also give us an option to control the size of the text in these boxes separate from everything else?

There are these settings: `editor.suggestFontSize` and `editor.suggestLineHeight`

**Comment by jrieken** [comment]
@MartinGC94 re https://github.com/microsoft/vscode/issues/29126#issuecomment-714782628 👉 https://github.com/microsoft/vscode/issues/99370

**Comment by rwcorbett** [comment]
re: [@MartinGC94 's comment](https://github.com/microsoft/vscode/issues/29126#issuecomment-714782628)
> Maybe it's an extension issue, but I don't see linebreaks for the different method overloads in Powershell. Here's how it looks in stable VS code:

Can we it set to break on word instead? This should respect the white-spaces better - something like `word-wrap: break-word;` or `overflow-wrap: break-word;` probably (I have not inspected with dev tools yet)

**Comment by jrieken** [comment]
This is done. Will ship with next stable and is already available in insiders. Please file bugs for further work and polish

**Comment by mikecann** [comment]
wooo! been following this for 3 years, excellent news ;)

**Comment by mbreslav** [comment]
Will the ability to resize the window also apply to the tooltip/hover window that is used by Hover providers?

**Comment by GustavoASC** [comment]
This feature is incredible and works perfectly. Thank you very much @jrieken for developing this :heart_eyes:
